src/test/java/uk/gov/gchq/palisade/policy/WrappedRuleTest.java

```diff
@@ -45,47 +47,47 @@ public void shouldNotConstruct0Arguments() {
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             // Then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
     }
 
     @Test
     public void shouldNotConstruct2Arguments() {
         try {
             //when
-            new WrappedRule<>(new TestRule(), o -> o.toString(), null);
+            new WrappedRule<>(new TestRule(), o -> o, null);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
         try {
             //when
             new WrappedRule<>(new TestRule(), null, o -> true);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
         try {
             //when
-            new WrappedRule<>(null, o -> o.toString(), o -> true);
+            new WrappedRule<>(null, (UnaryOperator<Object>) Object::toString, o -> true);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
     }
```

Comment:
Should really be done with the @Test(expected = xxxException.class), but i can see why its different here...
I wouldn't expect to try to check the exception message though, probably just leave it as "catch IllegalArgumentException, anything else will fail the test" as it is already doing, just drop the assertEquals

Reply:
Given all scenarios output the same message, I don't think it is important to validate the message and therefore we are just validating that the IllegalArgumentException is thrown for each scenario. Therefore this needs breaking down into the 3 different tests each with the @test(expected = xxxException.class) annotation.